src/redux/propertytree/propertyTreeMiddleware.ts

```diff
@@ -1,29 +1,65 @@
-import { createAction, Dispatch, UnknownAction } from '@reduxjs/toolkit';
+import { createAction, createAsyncThunk } from '@reduxjs/toolkit';
 
 import { api } from '@/api/api';
 import { onOpenConnection } from '@/redux/connection/connectionSlice';
-import { refreshGroups } from '@/redux/groups/groupsSlice';
 import type { AppStartListening } from '@/redux/listenerMiddleware';
-import { Property, PropertyMetaData, PropertyOwner } from '@/types/types';
+import {
+  Properties,
+  Property,
+  PropertyMetaData,
+  PropertyOwner,
+  PropertyOwners
+} from '@/types/types';
 import { rootOwnerKey } from '@/util/keys';
 
+import { refreshGroups } from '../groups/groupsSliceMiddleware';
+
+import { clearProperties, removeProperties } from './properties/propertiesSlice';
 import {
-  addProperties,
-  clearProperties,
-  removeProperties
-} from './properties/propertiesSlice';
-import {
-  addPropertyOwners,
   clearPropertyOwners,
   removePropertyOwners
 } from './propertyowner/propertyOwnerSlice';
 
-export const reloadPropertyTree = createAction<void>('reloadPropertyTree');
-export const propertyTreeWasChanged = createAction<void>('propertyTreeWasChanged');
-export const addUriToPropertyTree = createAction<{ uri: string }>('addUriToPropertyTree');
+export const reloadPropertyTree = createAction<void>('propertyTree/reload');
 export const removeUriFromPropertyTree = createAction<{ uri: string }>(
-  'removeUriFromPropertyTree'
+  'propertyTree/removeUri'
+);
+
+export const addUriToPropertyTree = createAsyncThunk(
+  'propertyTree/addUri',
+  async (uri: string) => {
```

Comment:
Imo we should explicitly type the return type here

Reply:
Perhaps on all of these async thunk functions

Reply:
Tried this. It was not straightforward... putting it in notion as to be done later